### ISSUE TITLE
Task Dependancy

### ISSUE THREAD

**Issue by ghost** [pleerock/gulpclass#4]
How do I need two write a task that depends on another task to be done first?

**Comment by pleerock** [comment]
Try to specify dependencies after task name.
`@Task(name?: string, dependencies?: string[])`

**Comment by ghost** [comment]
ok thank you. Can you maybe formulate an example just to make it more visual for me? Oh wait now I see it I guess.

**Comment by ghost** [comment]
Ok with the current gulpclass the dependencies are executed although the task ran already so dependancies do extra work instead of waiting for a task to finish. This actually worked in v0.0.2

**Comment by pleerock** [comment]
what do you mean? Do you want to say that your tasks are getting running in _parallel_ ? If so, then they should run in parallel. If you want to run the in sequence you should use `@SequenceTask` decorator.

**Comment by ghost** [comment]
I mean if I have a sequenceTask and it contains tasks that have dependencies, those dependencies are executed although they might have happend already in the sequenceTask before that task, so they run more often than necessary.

As far as I understand dependencies, the important thing is that they run before the Task that has the dependency runs. So if they are a part of a sequenceTask there should be a check if the dependency is already met when its the task's turn to run.

**Comment by pleerock** [comment]
Not sure I get you completely. `@SequenceTask` is simply using [run-sequence package](https://www.npmjs.com/package/run-sequence), and does not append anything to it. So its behaviour should not differ from that package. If you think there is a bug in @SequenceTask behaviour then try to describe your problem, or point to code where exactly you think is error, or wrong behaviour.
